sdk/identity/azure-identity/azure/identity/_credentials/azd_cli.py

```diff
@@ -194,7 +194,7 @@ def _run_command(command: str, timeout: int) -> str:
         # Fallback check in case the executable is not found while executing subprocess.
         if ex.returncode == 127 or ex.stderr.startswith("'azd' is not recognized"):
             raise CredentialUnavailableError(message=CLI_NOT_FOUND)
-        if "not logged in, run `azd auth login` to login" in ex.stderr:
+        if "not logged in, run `azd auth login` to login" in ex.stderr and "AADSTS" not in ex.stderr:
```

Comment:
Will this leak PII information?

Should we do more to sanitize the stderr in the case that "AADSTS" is present?

Reply:
As far as I am aware, PII shouldn't be leaked. I think the main note was that stderr was a bit verbose with the trace and correlation IDs.

We already do the token sanitization before where we raise the ClientAuthentication below. I also don't see other languages sanitizing the cli stderr in any additional manner. (e.g., [.NET](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/AzureCliCredential.cs#L166)).

Reply:
We only sanitize token. How about PII? e.g. user name?

Not sure if https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/AzureCliCredential.cs#L157-L159 is good enough.

We need to double check with .net.

Reply:
@christothes I know that .NET also does this [type of check](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/AzureCliCredential.cs#L157C22-L157C35). Do you think additional santization would be needed here? I'm uncertain if there are cases where PII would be in the AADSTS error message.

Reply:
I don't think there are any guarantees that AADSTS errors won't contain any identifiable data, but typically it would be OID such as a tenantId. I think the reason we special cased those errors is because typically they contain more diagnosable information than our standard errors. Trying to abstract them ends up hiding important information for troubleshooting.

Reply:
@xiangyan99 Since the main motivation for this is to improve diagnosibility, do you think it's fine to follow in .NET's footsteps with this change?

Reply:
Sounds good to me.

Reply:
Given this is for diagnosibility, should we enable it in debug log rather than error messages?

Reply:
Hmm, supposedly we could, but then the current NOT_LOGGED_IN error message would still be confusing unless a user turns on debug.

Currently, any AADSTS error messages that are returned without containing the "not logged in..." string are propagated to the user through the [error message](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/azd_cli.py#L145). Don't think we need to special case print the error to debug log in the case of "AADSTS" and "not logged in..." both being in the message.